### ISSUE TITLE
When disabling AAQ on a namespace gated pods are not being released

### ISSUE THREAD

**Issue by iholder101** [kubevirt/application-aware-quota#69]
**Context**:
A namespace can be removed from AAQ's targeted namespaces by:
* Modifying the namespace selector defined on that CR.
* Modifying the labels on a namespace (e.g. remove the `application-aware-quota/enable` from a namespace).

**What happened**:
When un-targeting a namespace:
* Pods under this namespace are gated forever.
* Objects created by AAQ are not being cleaned up.

In other words, the admin is responsible to clean up after AAQ, and AAQ does not fully honor its desired state.

**What you expected to happen**:
In such a circumstance, as a user (=admin), I would assume that AAQ won't be targeting this namespace anymore. This means that:
* All gated pods on that namespace should be released and scheduled.
* All of the objects created under this namespace would be removed.

So, TLDR: AAQ should cleanup after itself and not touch the non-targeted namespaces in any way.

**How to reproduce it (as minimally and precisely as possible)**:
1. Create AAQ with the default selector (`application-aware-quota/enable`)
2. Create a namespace and add this label to it.
3. Create an Application Resource Quota that's very restrictive.
4. Create a pod.
5. Watch that the pod is gated for scheduling.
6. Remove the label from the namespace / change namespace selector.
7. The pod is gated forever.

**Comment by iholder101** [comment]
FYI @Barakmor1 @vladikr

**Comment by Barakmor1** [comment]
The admin can to either delete the Quota or increase it to release gated pods. I don't believe we should clean the gated pods or monitor matching namespaces.

In my opinion, documenting that `AAQ.spec.namespaceSelector` only affects pod mutation at creation time is sufficient. I'm not convinced it will improve user experience for admins.

But if there is wide agreement to clean gated pods when` AAQ.spec.namespaceSelector doesn't match a namespace because of namespace modifications or `AAQ.spec.namespaceSelector` modifications , I guess it is possible to make the change.

@vladikr WDYT?

**Comment by dshchedr** [comment]
just a proposal - is it possible to prevent creation of ARQ in that namespaces?

it will also cover another issue - ARQ creates ResourceQuota for non-schedulable resources even if namespace does not have label
